File: frontend/src/schema/lead.ts
import { Company } from './company';

export interface Lead {
  id: string;
  company: Company;
  status: string;
  createdAt: Date;
  updatedAt: Date;
  ceoName: string;
  ceoLinkedinUrl: string;
  ceoEmail: string;
  ceoPhone: string;
  metadata: Record<string, any>;
}

File: backend/app/schema/lead.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Lead(BaseModel):
    id: str
    company_id: str
    status: str
    created_at: datetime
    updated_at: datetime
    ceo_name: Optional[str] = None
    ceo_linkedin_url: Optional[str] = None
    ceo_email: Optional[str] = None
    ceo_phone: Optional[str] = None
    metadata: Optional[dict] = None